src/reynir_correct/pattern.py

```diff
@@ -178,15 +178,37 @@ def __init__(self, ann: List[Annotation], sent: Sentence) -> None:
     def get_wordform(self, lemma: str, cat: str, variants: Iterable[str]):
         """ Get correct wordform from BinPackage, 
         given a set of variants """
-
-        # Get rid of argument variants in verbs:
-        realvars = [v for v in variants if not v.isdigit() and v not in SKIPVARS]
+        cases = ["nf", "þf", "þgf", "ef"]
+        if cat == "so":
+            # Get rid of argument variants in verbs:
+            realvars: Iterable[str]  = [v for v in variants if not v.isdigit() and v not in SKIPVARS]
+            if not "lh" in realvars:
+                realvars = [v for v in realvars if v not in cases]
+        else:
+            realvars= variants
         wordforms = BIN.lookup_variants(lemma, cat, tuple(realvars))
         if not wordforms:
             return ""
         # Can be many possible word forms, want the first one in most cases
         return wordforms[0].bmynd
 
+    def wrong_subtree(self, match: SimpleTree, so: SimpleTree) -> bool:
+        iptree = match.first_match("IP")
+        if not iptree:
+            return False
+    
+        for x in iptree.descendants:
+            if not x.is_terminal and x.tag == "IP":
+                if self.is_subtree(so, x):
```

Comment:
Is this not always True? The descendants of a tree are always subtrees of that tree.

Reply:
self.is_subtree(so, x) checks if the verb subtree we found (so) is under the right IP.
Added a docstring to explain better. 
We could have this tree:
S
    IP-1
        NP
        VP
            so_fh-1
            CP-REL
                IP-2
                    so_fh-2
Our search may match for IP-1 and so_fh-2. To make sure the rule doesn't fire, we check if we find any other IPs under IP-1. If we do, we check if so_fh-2 is a subtree of that IP. If so, we don't do anything.